File: js/paddle.js
function Paddle() {
  Entity.call(this);

  this.width  = 20;
  this.height = 100;

  this.y = game.height / 2 - this.height / 2;

  this.score = 0;
}

Paddle.prototype = Object.create(Entity.prototype);
Paddle.prototype.constructor = Paddle;

Paddle.prototype.update = function() {
  Entity.prototype.update.apply(this, arguments);

  this.y = Math.min(Math.max(this.y, 0), game.height - this.height);
};

File: app/models/paddle.rb
class Paddle < Entity

  attr_accessor :score

  def initialize
    super

    @width = 20
    @height = 100

    @y = @game.height / 2 - @height / 2

    @score = 0
  end

  def update
    super

    @y = [[@y, 0].max, @game.height - @height].min
  end

end
